packages/core/admin/server/src/services/statistics.ts

```diff
@@ -0,0 +1,425 @@
+  locale: string | null;
+  createdBy: User;
+  updatedBy: User;
+  strapi_assignee?: object | null;
+  strapi_stage?: StrapiStage | null;
+};
+
+export interface PieData {
+  name: string;
+  value: number;
+  entries?: Entry[];
+}
+
+type Locales = {
+  [key: string]: Entry[];
+};
+
+export type KeyNumbers = {
+  [key: string]: number;
+};
+
+interface KeyNumberProvider {
+  name: string;
+  resolve(): Promise<number>;
+}
+
+export type ContentTypeStatistics = {
+  locales: Locales;
+  count: number;
+  modifiedCount: number;
+  statusPieChart: PieData[];
+  localePieChart: PieData[];
+  topContributors: User[];
+  latestDraftEntries: Entry[];
+  latestPublishedEntries: Entry[];
+  stackedAreaChartLocales: Array<{ [key: string]: string }>;
+  stackedAreaChartStages: Array<{ [key: string]: string }>
+  assignedEntriesPieChart: PieData[];
+  assignedEntriesCount: number;
+  stackedAreaChartAssignedStages: Array<{ [key: string]: string }>;
+};
+
+export interface User {
+  id: number;
+  username: string;
+  firstname: string;
+  lastname: string;
+}
+
+export interface Payload {
+  entry: {
+    documentId: string;
+    locale?: string;
+  };
+  uid: string;
+}
+
+export type Activity = {
+  action: string;
+  createdAt: string;
+  date: string;
+  documentId: string;
+  id: number;
+  locale: string;
+  payload: Payload;
+  publishedAt: string;
+  updatedAt: string;
+  user: User;
+};
+
+export type Release = {
+  createdAt: string;
+  documentId: string;
+  id: number;
+  locale: string;
+  name: string;
+  publishedAt: string;
+  releasedAt: string;
+  scheduledAt: string;
+  status: string;
+  timezone: string;
+  updatedAt: string;
+};
+
+export type EEStatistics = {
+  upcomingReleases: Release[] | Array<object>;
+  latestActivities: Activity[] | Array<object>;
+  upcomingReleasesCount: number,
+  workflowsCount: number,
+};
+
+const providers: Map<string, KeyNumberProvider> = new Map();
+
+export function registerKeyNumbersProvider(provider: KeyNumberProvider): void {
+  if (providers.has(provider.name)) {
+    throw new Error(`Provider with name ${provider.name} already exists`);
+  }
+
+  providers.set(provider.name, provider);
+}
+
+export async function getKeyNumbers(): Promise<KeyNumbers> {
+  const keyNumbers: KeyNumbers = {
+    entries: 0,
+    assets: 0,
+    contentTypes: 0,
+    components: 0,
+    locales: 0,
+    admin_users: 0,
+    webhooks: 0,
+    apiTokens: 0
+  };
+
+  const results = Array.from(providers.values()).map((provider) =>
+    provider.resolve().then((result) => {
+      keyNumbers[provider.name] = result;
+    })
+  );
+
+  await Promise.all(results);
+
+  return keyNumbers;
+}
+
+export async function getContentTypeStatistics(contentType: UID.CollectionType, userId: number): Promise<ContentTypeStatistics> {
+  // Fetch entries with associated user data that would always be there and just enough for the dashboard
+  const [entries, initialCount] = await strapi.db.query(contentType).findWithCount({
```

Comment:
While the documentService doesn't have a findWidthCount we can still use findMany and count here since we aren't saving any resources on using findWithCount. That function makes two DB calls on it's own: https://github.com/strapi/strapi/blob/a7ba501bdf32bc14553ec65e9037a1b8b6f20915/packages/core/database/src/entity-manager/entity-repository.ts#L47-L52

Reply:
Or we sweet talk the engineers into adding a findWithCount function the document service

Reply:
on top of this I would add that we might want to exclude drafts or published versions here, or a pair of them might count as 2 entries, using the doc service would handle that automatically